lib/validation.js

```diff
@@ -234,7 +235,17 @@ module.exports = function validation (yargs, usage, y18n) {
       })
     } else {
       yargs.global(key)
-      implied[key] = value
+      if (!implied[key]) {
+        implied[key] = []
+      }
+      if (Array.isArray(value)) {
+        value.forEach((i) => self.implies(key, i))
+      } else if (typeof value === 'string') {
+        implied[key].push(value)
+      } else {
+        throw new YError('Invalid second argument passed to implied.' +
```

Comment:
the reason that tests are failing is that there's this weird edge-case feature where an argument can imply a certain # of positional arguments in `_`:

```js
yargs.implies({
  foo: 4
})
```

So the second argument can be a number, or a string, or an array with your design. Rather than pulling in the `YError` object and throwing your own error, I would pull in the `argsert` library and do this:

```js
  self.implies = function implies (key, value) {
    argsert('<string|object> [array|number|string]', [key, value], arguments.length)

    if (typeof key === 'object') {
      Object.keys(key).forEach((k) => {
        self.implies(k, key[k])
      })
    } else {
      yargs.global(key)
      if (!implied[key]) {
        implied[key] = []
      }
      if (Array.isArray(value)) {
        value.forEach((i) => self.implies(key, i))
      } else {
        implied[key].push(value)
      }
    }
  }```